src/main/java/com/onarandombox/multiverseinventories/share/Sharables.java

```diff
@@ -582,6 +590,149 @@ public boolean updatePlayer(Player player, PlayerProfile profile) {
+                        intersection.retainAll(completedAdvancements);
+
+                        for (String criteria: revoke) {
+                            processedAdvancements.add(criteria);
+                            completedAdvancements.remove(criteria);
+                            player.getAdvancementProgress(advancement).revokeCriteria(criteria);
+                        }
+                        for (String criteria: intersection) {
+                            processedAdvancements.add(criteria);
+                            completedAdvancements.remove(criteria);
+                            player.getAdvancementProgress(advancement).awardCriteria(criteria);
+                        }
+
+                        // here's the idea: revoke all (criteria \ completedAdvancements), grant (criteria intersect completedAdvancements)
+                        // also, we don't need to grant/revoke anything in processedAdvancements since they've already been granted/revoked!
+                    }
+                    if (announceAdvancements) player.getWorld().setGameRule(GameRule.ANNOUNCE_ADVANCEMENTS, true);
+                    return advancements != null;
+                }
+            }).defaultSerializer(new ProfileEntry(false, "advancements")).altName("achievements").build();
+
+    /**
+     * Sharing Statistics.
+     */
+    public static final Sharable<Map> GAME_STATISTICS = new Sharable.Builder<Map>("game_statistics", Map.class,
+            new SharableHandler<Map>() {
+                @Override
+                public void updateProfile(PlayerProfile profile, Player player) {
+                    Map<String, Integer> playerStats = new HashMap<>();
+                    for (Statistic stat: Statistic.values()) {
+                        if (stat.getType() == Statistic.Type.UNTYPED) {
+                            int val = player.getStatistic(stat);
+                            // no need to save values of 0, that's the default!
+                            if (val != 0) playerStats.put(stat.name(), val);
+                        }
+                    }
+                    profile.set(GAME_STATISTICS, playerStats);
+                }
+
+                @Override
+                public boolean updatePlayer(Player player, PlayerProfile profile) {
+                    Map<String, Integer> playerStats = profile.get(GAME_STATISTICS);
+                    for (Statistic stat : Statistic.values()) {
+                        if (stat.getType() == Statistic.Type.UNTYPED) player.setStatistic(stat, 0);
+                    }
+                    if (playerStats == null) {
+                        return false;
+                    }
+                    for (String stringStat : playerStats.keySet()) {
+                        Statistic stat = Statistic.valueOf(stringStat);
+                        if (stat.getType() == Statistic.Type.UNTYPED) player.setStatistic(stat, playerStats.get(stat.name()));
+                    }
```

Comment:
Maybe iterating through the entry set is better?
```java
for (Map.Entry<String, Integer> statInfo : playerStats.entrySet()) {
    Statistic stat = Statistic.valueOf(statInfo.getKey());
    if (stat.getType() == Statistic.Type.UNTYPED) {
        player.setStatistic(stat, statInfo.getValue());
    }
}
```

Reply:
Sure, this seems better. Didn't know you could do this in Java 😝.